index.html

```diff
@@ -656,29 +656,74 @@ <h4>Registration Information</h4>
                             </tr>
                         </thead>
                         <tbody>
-                            <tr class="rfc2119-table-assertion" id="tdd-vocab-created--Thing">
+                            <tr class="rfc2119-table-assertion" id="discovery-vocab-created--RegistrationInformation">
                                 <td><code>created</code></td>
                                 <td>Provides information when the TD instance was
                                     created inside the directory.</td>
                                 <td>system-generated (read-only)</td>
                                 <td><a href="http://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/#dateTime"><code>dateTime</code></a></td>
                             </tr>
-                            <tr class="rfc2119-table-assertion" id="tdd-vocab-modified--Thing">
+                            <tr class="rfc2119-table-assertion" id="discovery-vocab-modified--RegistrationInformation">
                                 <td><code>modified</code></td>
                                 <td>Provides information when the TD instance was
                                     last modified inside the directory.</td>
                                 <td>system-generated (read-only)</td>
                                 <td><a href="http://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/#dateTime"><code>dateTime</code></a></td>
                             </tr>
-                            <!-- <tr class="rfc2119-table-assertion" id="tdd-vocab-expires--Thing">
+                            <tr class="rfc2119-table-assertion" id="discovery-vocab-expires--RegistrationInformation">
                                 <td><code>expires</code></td>
                                 <td>Provides information when the TD instance registration will
                                     expire.</td>
                                 <td>optional</td>
                                 <td><a href="http://www.w3.org/TR/2012/REC-xmlschema11-2-20120405/#dateTime"><code>dateTime</code></a></td>
-                            </tr> -->
+                            </tr>
                         </tbody>
                     </table>
+                    
+                    <p>
+                        The `created` and `updated` attributes are set by the directory
+                        and returned to consumers in <a>Enriched TD</a>s to allow
+                        bookkeeping and search.
+                        <span class="rfc2119-assertion" id="tdd-registrationinfo-created-updated">
+                            The maintenance of creation and modification time of <a>TD</a>s inside
+                            the directory are OPTIONAL for servers.
+                        </span>
+                    </p>
+                    
+                    <p>
+                        The `expires` attribute is set by the producer and submitted to the directory
```

Comment:
The directory may also have a maximum value that it may be set to, and may provide a default.  So really the registration response should be the final result: the producer "asks" for a certain expiry, the directory either accepts it or uses a possibly shorter value, then returns the result (e.g. in an enriched TD).   The directory should also be able to set a finite expiry value even if the requester does not specify (and then it is (always) the responsibility of the the requester to re-register to update the expiry before the TD expires).

In other words, TTL should always be finite...

Reply:
I agree that making registrations time-limited is useful in most scenarios, however, making it mandatory while not prescribing a maximum value defeats its purpose. Servers could set it to e.g. 10 years by default which is useless. On the other hand, we can't decide on a maximum value that is appropriate for all applications.